packages/react-components/theme-designer/src/components/Sidebar/Sidebar.tsx

```diff
@@ -97,52 +94,92 @@ export const Sidebar: React.FC<SidebarProps> = props => {
   const styles = useStyles();
 
   const [tab, setTab] = React.useState<TabValue>('use');
-  const handleTabChange = React.useCallback(
-    (event, data) => {
-      props.setTheme('Custom');
-      setTab(data.value);
-    },
-    [props],
-  );
+  const handleTabChange = (event: SelectTabEvent, data: SelectTabData) => {
+    setTheme('Custom');
+    dispatchCustom(custom);
+    setTab(data.value);
+  };
 
   const keyColorId = useId();
-  const handleOnChange = React.useCallback(e => props.setKeyColor(e.target.value), [props]);
-
   const hueTorsionId = useId();
-  const handleHueTorsionChange = React.useCallback(e => props.setHueTorsion(e.target.value / 10), [props]);
-
   const lightCpId = useId();
-  const handleLightCpChange = React.useCallback(e => props.setLightCp(e.target.value / 100), [props]);
-
   const darkCpId = useId();
-  const handleDarkCpChange = React.useCallback(e => props.setDarkCp(e.target.value / 100), [props]);
-
   const themeId = useId();
-  const handleThemeChange: MenuProps['onCheckedValueChange'] = React.useCallback(
-    (e, data) => {
-      props.setTheme(data.checkedItems[0] as string);
-    },
-    [props],
-  );
 
-  const Use = React.memo(() => (
-    <div className={styles.content} role="tabpanel" aria-labelledby="Use">
-      <div className={styles.inlineInputs}>
+  const [theme, setTheme] = React.useState<string>('Teams Light');
+
+  const initialCustom = {
+    keyColor: '#006bc7',
+    hueTorsion: 0,
+    darkCp: 2 / 3,
+    lightCp: 1 / 3,
+    isDark: false,
+  };
+
+  const customReducer = (state: CustomAttributes, action: CustomAttributes) => {
```

Comment:
this is an area we can improve. with only a single reducer return, this is basically just useState. Would be nice to get a better pattern here to reduce the amount of complex code below. fine for now